### PR TITLE
Update to Vega-Lite 5.15.1

### PR DESCRIPTION
**Pull request by joelostblom**
@mattijn @binste @jonmmease @ChristopherDavisUCI Would you be OK if we make a 5.1.2 (this seems to small for 5.2?) early next week to update VL once https://github.com/vega/vega-lite/pull/9106 is merged and to include https://github.com/altair-viz/altair/pull/3191? It would be helpful for a class I'm teaching mid-next week. I can try to make the release since I'm the one who needs it, but if anyone has time to help if needed it is of course appreciated (the release instructions seem straightforward so hopefully I don't set anything on fire).

I put up this PR as a draft to make sure I am doing the correct thing when updating VL; I will flag for review when there is an actual VL 5.16 release. I bumped the version number in `tools/generate_schema_wrapper.py` and then ran `python tools/generate_schema_wrapper.py`, and then `hatch test`. Is that all?

### CODE REVIEW

File: CONTRIBUTING.md
Comment (joelostblom) [inline CONTRIBUTING.md:76]: Is this needed to be done manually, or does our workflow do it automatically and add it to the PR?

File: RELEASING.md
Comment (joelostblom) [inline RELEASING.md:2]: This is old right? We don't need altair_saver for the release anymore?

Also @jonmmease When I run `hatch test`, I run into an issue because vl-convert does not yet support the Vega-Lite version string. Does that mean we should wait for a vl-convert release before updating altair, so the order of releases needs to be `Vega-Lite` -> `vl-convert` -> `altair`?

File: altair/vegalite/v5/schema/core.py
Comment (joelostblom) [inline altair/vegalite/v5/schema/core.py:7063]: As mentioned in https://github.com/vega/vega-lite/pull/9066#issuecomment-1741211330, this is incorrect and will likely be removed in 5.16

File: doc/releases/changes.rst
Comment (joelostblom) [inline doc/releases/changes.rst:18]: I will update this date depending on the actual release

File: RELEASING.md
Comment (jonmmease) [reply RELEASING.md:2]: Yes, exactly. I'll watch for the Vega-Lite release and then publish vl-convert right after

Comment (jonmmease) [reply RELEASING.md:2]: Then you'll want to update the vl-convert-python version string at:

https://github.com/altair-viz/altair/blob/5a7814cd14a2b336bea6e5b2bb7391467afe88fc/altair/utils/_importers.py#L32

It will probably be 0.14.0, but if Vega get's released as well I may bump it up to 1.0.0

**Comment by jonmmease** [comment]
I'm supportive of doing a 5.1.2 release early next week. I haven't gone through the release process, but happy to make sure vl-convert is ready once Vega-Lite 5.16 is out.

File: CONTRIBUTING.md
Comment (binste) [reply CONTRIBUTING.md:76]: This is already documented in https://github.com/altair-viz/altair/blob/main/NOTES_FOR_MAINTAINERS.md so I think we don't need this section :) But maybe you could link to that markdown file here so it's easier to spot?

**Review by binste — COMMENTED** [review]
Sounds good to me too, thanks for offering to do the release! Feel free to ping me again in case you need a review once the new VL version is out.

File: CONTRIBUTING.md
Comment (binste) [reply CONTRIBUTING.md:76]: And yes, it needs to be done manually so that these files can be reviewed as well. In the pipeline/workflow, we then again run the `generate_schema_wrapper.py` to see if it does not produce any additional changes which have not already been committed to make sure that the generated code is always in sync with what is defined in `tools`.

File: tools/generate_schema_wrapper.py
Comment (binste) [inline tools/generate_schema_wrapper.py:33]: Just a heads-up that I'm simplifying this part of the code in #3207 and will merge it soon -> You will have a merge conflict here which should be simple to resolve. You can then just set `SCHEMA_VERSION = "v5.15.0"` as we no longer need the dictionary.

File: CONTRIBUTING.md
Comment (joelostblom) [reply CONTRIBUTING.md:76]: Ah yes, there is a maintainers file as well =p I forgot about that one; I will remove this section.

Comment (joelostblom) [reply CONTRIBUTING.md]: I kept a short section to indicate that the schema generation should be run after any changes by pointed to the notes for Vega-Lite updates

**Comment by mattijn** [comment]
Go ahead! I'm a bit occupied these days. Ping me if you get stuck in the release proces.

**Comment by ChristopherDavisUCI** [comment]
Hi all, I'm going to try to start being more involved with the day-to-day Altair maintenance than I have been in the last few months.  Many things are still mysterious to me, but especially please ping me if there's something in particular you think I might be able to help with.

**Comment by joelostblom** [comment]
v5.15.1 was just released, when I tried updating this PR it fails to find a URL required the generate the schema. The URL is  https://vega.github.io/schema/vega-lite/v5.15.1.json which indeed seems to be missing so maybe there is just a build process that takes some time for that to come up?

**Comment by binste** [comment]
The publishing step to vega.github.io/schema failed, I opened [an issue ](https://github.com/vega/schema/issues/9). However, the schema already exists [on GitHub](https://raw.githubusercontent.com/vega/schema/master/vega-lite/v5.15.1.json) so you could point the Altair schema generation temporarily to that URL to get a head start until someone gets a chance to fix it :)

**Comment by jonmmease** [comment]
vl-convert-python 0.14.0 was just published to PyPI with Vega-Lite 5.15.1 support

File: RELEASING.md
Comment (jonmmease) [reply RELEASING.md:2]: It is indeed 0.14.0

Comment (jonmmease) [reply RELEASING.md:2]: Also, reminder to make this update in `_importers.py`.

**Comment by joelostblom** [comment]
All checks are passing so marking this as ready for review!

File: RELEASING.md
Comment (joelostblom) [reply RELEASING.md:2]: Thank you, updated!

File: doc/releases/changes.rst
Comment (joelostblom) [inline doc/releases/changes.rst:22]:
```suggestion
  see `Vega-Lite Release Notes <https://github.com/vega/vega-lite/releases>`_.
- Updated vl-convert from version 0.13.0 to version 0.14.0;
  see `vl-convert Release Notes <https://github.com/vega/vl-convert/releases>`_.
  - Notably, this release makes it possible to save Altair charts as PDFs via `chart.save('filename.pdf')`
```

@jonmmease I realize we're highlighting the bumps in Vega-Lite version and was thinking maybe we should do the same for VegaFusion and vl-convert? At least when they add new features, such as 0.14 adding support for saving charts as PDFs, which is something I definitely think we should highlight in the release notes for Altair somehow.

Comment (joelostblom) [reply doc/releases/changes.rst:22]: Ok that sounds great. Having a 5.2 release for this improvement feels appropriate!

**Review by binste — APPROVED** [review]
Looks good to me! Thank you @joelostblom and @jonmmease

**Comment by joelostblom** [comment]
5.1.2 released! It was scary but I think everything is working. I didn't encounter any errors, and I tried installing 5.1.2 myself from pip and it seems to work as expected. Just waiting for https://github.com/conda-forge/altair-feedstock/pulls to update and then that should be it

**Comment by binste** [comment]
Looks good, thanks! 😃

**Pull request merged**
